testdata/src/go.uber.org/functioncontracts/inference/functioncontracts-with-inference.go

```diff
@@ -0,0 +1,125 @@
+//  Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// <nilaway contract enable>
+package inference
+
+import "math/rand"
+
+// Test the contracted function contains a full trigger nilable -> return 0.
+// contract(nonnil -> nonnil)
+func fooReturn(x *int) *int { // want "^ Annotation on Result 0.*\n.*Must be NILABLE.*\n.*AND.*\n.*Must be NONNIL.*NONNIL$"
```

Comment:
We don't have `a2` or `b2` in the error message, so the best thing we can do now is to directly put the line number, which, however, prevents future update of the test file.

Maybe we can have intermediate variable names when we have the new format of error messages? @sonalmahajan15

Reply:
Yes, the variable names would likely be a part of the error message. I can see `a2` would definitely be, in the form of `unassigned variable a2`

Reply:
Ok, let's make a small issue for a follow up fix here to include the variable name after Sonal's change lands. It will likely be a one line PR, but that's fine 🙂 (Issue can be a title and a link to this thread, no need to describe in detail)

Reply:
See issue https://github.com/uber-go/nilaway/issues/35.